src/types/Document.ts

```diff
@@ -27,6 +27,152 @@ export type Document = {
   archived_file_name: string | null;
 };
 
+export type DocumentFilter = {
+  /** Title of document starts with */
+  title__istartswith: string;
+  /** Title of document ends with */
+  title__iendswith: string;
+  /** Title of document contains */
+  title__icontains: string;
+  /** Tilte of document is exactly */
+  title__iexact: string;
+  /** Title or content of document contains */
+  title_content: string;
+  /** Content of document starts with */
+  content__istartswith: string;
+  /** Content of document ends with */
+  content__iendswith: string;
+  /** Content of document contains */
+  content__icontains: string;
+  /** Content of document is exactly */
+  content__iexact: string;
+  /** Archive serial number of document is */
+  archive_serial_number: number;
+  /** Archive serial number of document is greater than */
+  archive_serial_number__gt: number;
+  /** Archive serial number of document is greater or equal than */
+  archive_serial_number__gte: number;
+  /** Archive serial number of document is lower than */
+  archive_serial_number__lt: number;
+  /** Archive serial number of document is lower or equal than */
+  archive_serial_number__lte: number;
+  /** Archive serial number of document is null */
+  archive_serial_number__isnull: boolean;
+  /** Created year of document is */
+  created__year: number;
+  /** Created month of document is */
+  created__month: number;
+  /** Created day of month of document is */
+  created__day: number;
+  /** Created date of document is greater than */
+  created__date__gt: string;
+  /** Created date of document is greater than */
+  created__date__lt: string;
+  /** Created is grater than */
+  created__gt: string;
+  /** Created is lower than */
+  created__lt: number;
+  /** Added year is */
+  added__year: number;
+  /** Added month is */
+  added__month: number;
+  /** Added day of month is */
+  added__day: number;
+  /** Added date is greater than */
+  added__date__gt: string;
+  /** Added date is lower than */
+  added__date__lt: string;
+  /** Added is greater than */
+  added__gt: string;
+  /** Added is lower than */
+  added__lt: string;
+  /** Modified year is */
+  modified__year: number;
+  /** Modified month is */
+  modified__month: number;
+  /** Modified day of month */
+  modified__day: number;
+  /** Modified date is greater than */
+  modified__date__gt: string;
+  /** Modified date is lower than */
+  modified__date__lt: string;
+  /** Modified is greater than */
+  modified__gt: string;
+  /** Modified is lower than */
+  modified__lt: string;
+  /** Correspondent is null */
+  correspondent__isnull: boolean;
+  /** Correspondent id is */
+  correspondent__id: CorrespondentId;
+  /** TODO: comma sperated list? arry or string?  Correspondent id is in */
+  correspondent__id__in: number[];
```

Comment:
if you do not know the type please make it `unknown`.
```suggestion
  correspondent__id__in: unknown;
```